src/corehost/cli/hostpolicy.cpp

```diff
@@ -275,7 +275,7 @@ int run(const arguments_t& args, pal::string_t* out_host_command_result = nullpt
     std::vector<char> managed_app;
     pal::pal_clrstring(args.managed_application, &managed_app);
 
-    breadcrumb_writer_t writer(&breadcrumbs);
+    breadcrumb_writer_t writer(breadcrumbs_enabled, &breadcrumbs);
     if (breadcrumbs_enabled)
```

Comment:
Any thoughts on removing this check here too?  Then you wouldn't need to assert in `begin_write`.

```C++
breadcrumb_writer_t writer(breadcrumbs_enabled, &breadcrumbs);
writer.begin_write();
```

`begin_write` would check if it is enabled, and skip doing anything if it isn't.

Reply:
Sure makes it a bit more consistent. Thanks